### PR TITLE
Fix service-type graph not showing service-to-service trace edges

### PR DESCRIPTION
**Pull request by jotak**
Fixes https://github.com/kiali/kiali/issues/3572

**Before**
Capture d’écran de 2021-01-11 12-09-05

**After**
Capture d’écran de 2021-01-11 12-05-51

### CODE REVIEW

**Comment by jotak** [comment]
Note, for seeing this kind of traces with both workload-to-service and service-to-service edges while testing, you can use demo-mesh-arena with traces enabled, see quickstart https://github.com/jotak/demo-mesh-arena/#quick-start

File: src/components/CytoscapeGraph/CytoscapeTrace.ts [hunk 1]
```diff
@@ -27,7 +27,25 @@ const showSpanSubtrace = (cy: Cy.Core, graphType: GraphType, span: Span) => {
...
+    if (sourceAppNs) {
+      let selector = `node[!${CyNode.isGroup}][${CyNode.nodeType}="${NodeType.SERVICE}"][${CyNode.app}="${sourceAppNs.app}"][${CyNode.namespace}="${sourceAppNs.namespace}"]`;
+      let parent = cy.elements(selector);
+      if (!parent || parent.length === 0) {
+        // Try workload
+        const sourceWlNs = searchParentWorkload(span);
+        if (sourceWlNs) {
+          selector = `node[${CyNode.workload}="${sourceWlNs.workload}"][${CyNode.namespace}="${sourceWlNs.namespace}"]`;
+          parent = cy.elements(selector);
+        }
```
Comment (aljesusg) [inline src/components/CytoscapeGraph/CytoscapeTrace.ts:42]: What do you think about this and avoid another if confition?

```ts
const sourceWlNs = searchParentWorkload(span);
if ((!parent || parent.length === 0) && sourceWlNs) {
        // Try workload
          selector = `node[${CyNode.workload}="${sourceWlNs.workload}"][${CyNode.namespace}="${sourceWlNs.namespace}"]`;
          parent = cy.elements(selector);
}
```
do you make sense to you?

**Review by aljesusg — APPROVED** [review]
Not block, just a comment LGFM anyway

File: src/components/CytoscapeGraph/CytoscapeTrace.ts (hunk 1 above)
Comment (aljesusg) [reply src/components/CytoscapeGraph/CytoscapeTrace.ts:42]: Note: I didn't check searchParentWorkload(span) , maybe it's a lot of complexity in that method

Comment (jotak) [reply src/components/CytoscapeGraph/CytoscapeTrace.ts:42]: Thanks for the review.
`searchParentWorkload` is neither very complex nor very trivial, but still, I'd prefer not to do unnecessary call here.

Comment (aljesusg) [reply src/components/CytoscapeGraph/CytoscapeTrace.ts:42]: understood so we can merge this one

Comment (jshaughn) [reply src/components/CytoscapeGraph/CytoscapeTrace.ts:42]: post-merge review :-)  The logic looks OK to me, I think it should work as expected.  Style-wise I may have used a switch statement but that's just personal pref.

Comment (aljesusg) [reply src/components/CytoscapeGraph/CytoscapeTrace.ts:42]: OMG Sorry my fault @jshaughn :'( I didn't see the others reviewers assign :(

Comment (jshaughn) [reply src/components/CytoscapeGraph/CytoscapeTrace.ts:42]: Not a problem, @aljesusg.   I usually wait for the PR developer to perform the merge, but it's not a strict process on our team.  Anyway, no big deal!

**Pull request merged**